File: LeetCode Practice Problems/102-Binary Tree Level Order Traversal.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        if(root == NULL)
            return ans;
        queue<TreeNode*> q;
        q.push(root);
        
        while(!q.empty()){
            queue<TreeNode*> t;
            vector<int> res;
            while(!q.empty()){
                TreeNode* r = q.front();
                q.pop();
                if(r->left != NULL)
                    t.push(r->left);
                if(r->right != NULL)
                    t.push(r->right);
                
                res.push_back(r->val);
            }
        
            ans.push_back(res);
            q = t;
        }
        
        return ans;
    }
};

File: LeetCode/LeetCode Practice Problems/102- Binary Tree Level Order Traversal.java
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> q = new LinkedList();
        
        if(root != null) {
            q.add(root);
        }
        
        List<List<Integer>> a = new ArrayList();
        
        while(!q.isEmpty()) {
            int sz = q.size();
            List<Integer> l = new ArrayList();
            while(sz > 0) {
                sz--;
                TreeNode t = q.remove();
                l.add(t.val);
                
                if(t.left != null) q.add(t.left);
                if(t.right != null) q.add(t.right);
            }
            
            a.add(l);
        }
        return a;
    }
}
